_posts/2018-10-01-why-developers-do-not-like-change.md

```diff
@@ -0,0 +1,75 @@
+
+ * Move house
+ * Change job
+ * Re-decorate a room
+ * Change your car
+
+## No one likes to lose
+From birth, it is pre-determined that we do not like losing. In the brilliant book 
+[Thinking, Fast and Slow](https://www.amazon.co.uk/Thinking-Fast-Slow-Daniel-Kahneman/dp/0141033576) talks about gains
+and losses. I won't delve into the details of Prospect Theory here, but if you want to understand more I suggest you
+read the book.
+
+In the book Kahneman explains how we react worse to losses than to gains, and we are much more sensitive to losses.
+This fear of losing can help explain why developers are not fond of organisational changes.
+
+For example, take the below scenario:
+ 
+
+> Company X has an software engineering department that consists of ten developers.
+
+
+Company X wish to make some changes, if you were one of those developers, which on of the below would you prefer?
+
+> A - Company X proposes a change in which there is a 70% chance you will lose your job
+
+> B - Company X proposes a change in which there is a 5% chance you will lose your job
+
+
+If you are like the majority of people, you will prefer option `B`, despite the fact that both options present potential
+losses (in this case, your job).
+
+Unfortunately, when changes are proposed the facts are not known and so you would not be able to make such a decision as
+above. There normally is a large amount of uncertainty, and your decision to remain at a company will be a gamble.
+
+## The illusion of moving
+When presented with an organisational change marked with much uncertainty, many think that moving company will be the
+solution. However, this is an illusion and can often result in people hastily jumping ship and then having to do the
+same at their new company because the grass was not in fact greener.
+
+## Creatures of habit
+As developers, we are creatures of habit. Much of the technologies we like is formed due to our familiarity with such
+technologies. For example, say your favourite programming language is Java, this will most likely be because you have
+a lot of experience using it and you have good memories programming in Java.
+
+Now say your bosses tell you that you must now use C# instead, this conflicts with your passion for Java. You most
+likely view this change as a loss, not a gain. Therefore you will most likely make a fuss and voice your dissatisfaction
+of using C#.
+
+## My advice to developers
+In the event of you finding yourself in the middle of a proposed organisational change, seek out as much information as
+possible to help you make your decision. See what you have to gain or lose from the change. If there is not much to lose
+then I would recommend seeing how the change goes, you might find yourself in a situation where you can learn a lot.
+
+## My advice to managers
+Understand that developers, like anyone else, are creatures of habit and will be skeptical of change. Listen to
```

Comment:
Another good piece of advice might be to enact change more democratically, to get feedback from developers early on. A mistake I often see is managers "pushing" change down from "up above" rather than treating change as a constant, iterative give-and-take with developers. 

It can also help to review changes. Scrum's idea of making small, experimental changes to processes frequently and reviewing them in retrospectives once every sprint to enable rapid iterative changes is one that I think is quite powerful. That way you can highlight that changes are experimental, that you can always undo them with minimal effort because all changes are reviewed so frequently, and that makes it easier to convince people to try them and more difficult to argue against because "at worst we do it for just one sprint". Plus, it's important to review changes you implement- I find high-level management thinking often falls into the trap of assuming that changes they make are good, and not reviewing what effect they've had.

Reply:
This is a good point actually.

Also, management often don't fully comprehend the changes that are taking place and how they will impact everyone involved.

They might have been sold an idea by someone, and have bought into it without really understanding it.